src/ripple/app/misc/impl/ValidatorSite.cpp

```diff
@@ -401,13 +400,18 @@ ValidatorSite::onSiteFetch(
                          res.result() == status::found              ||
                          res.result() == status::temporary_redirect)
                 {
-                    newLocation = processRedirect (res, siteIdx, lock_sites);
+                    auto newLocation =
+                        processRedirect (res, siteIdx, lock_sites);
+                    assert(newLocation);
```

Comment:
This assert seems unnecessary as `newLocation` returns a pointer or throws on failure.